### DISCUSSION TITLE
Solution similar to `"compilerOptions": { "types": ["bun-types"] }` for Deno for Node+Deno projects?

### DISCUSSION THREAD

**Discussion by tordans** [denoland/deno#21237] [Q&A]
**Kontext:** I have a NextJS Project with a `scripts` folder. I want to use Deno to run the scripts in this folder and use things like `Deno.writeTextFile`.

**Issue:** Right now, `Deno` is unkown ("Cannot find name 'Deno'. — ts(2304)").

**What I tried:** I tried adding `/// <reference lib="deno.ns" />` to the script file or a `types.d.ts` in that folder or to import this types file in my script file … all don't solve the VS Code TS error.

**What I am looking for:** Right now, I have Bun configured in the same way (NextJS with scripts folder which uses Bun). The types are working with `"compilerOptions": { "types": ["bun-types",  "web"] }` in my project `tsconfig.json`.

_How do I solve this in Deno land?_

Right now, it looks like that  multi-root workspace as described by this FAQ https://docs.deno.com/runtime/manual/advanced/typescript/faqs#how-do-i-combine-deno-code-with-non-deno-code-in-my-ide are the only way to solve this in Deno land. However, that is very VS Code specific and also seems like a complex config since my script folder is part of the main project (so I would need to include/exclude this folder for the two workspaces?)… — So, is there a different way?

**Comment by bcheidemann** [comment]
In addition to @kazunetakeda25's excellent answer, if you are using the Deno VS Code plugin, you can use `deno.enablePaths`:

```json
// .vscode/settings.json
{
  "deno.enablePaths": [
    "scripts"
  ]
}
```
